src/dotnet/APIView/APIViewWeb/Repositories/CosmosReviewRepository.cs

```diff
@@ -141,18 +141,65 @@ public async Task<IEnumerable<string>> GetReviewFirstLevelPropertiesAsync(string
             {
                 var searchAsQueryStr = ArrayToQueryString<string>(search);
                 var searchAsSingleString = '"' + String.Join(' ', search) + '"';
-                queryStringBuilder.Append($" AND (r.Author IN {searchAsQueryStr}");
-                queryStringBuilder.Append($" OR STRINGEQUALS(r.Revisions[0].Name, {searchAsSingleString}, true)");
-                queryStringBuilder.Append($" OR CONTAINS(r.Name, {searchAsSingleString}, true)");
-                queryStringBuilder.Append($" OR CONTAINS(r.ServiceName, {searchAsSingleString}, true)");
-                queryStringBuilder.Append($" OR CONTAINS(r.PackageDisplayName, {searchAsSingleString}, true)");
-                queryStringBuilder.Append($" OR (CONTAINS(r.Name, \"{search[0]}\", true)");
-
-                for (int i = 1; i < search.Count; i++)
+
+                var hasExactMatchQuery = search.Any(
+                    s => (
+                    s.StartsWith("package:") || 
+                    s.StartsWith("pr:") || 
+                    s.StartsWith("author:") || 
+                    s.StartsWith("service:") ||
+                    s.StartsWith("name:")
+                ));
+
+                if (hasExactMatchQuery)
+                {
+                    foreach (var item in search)
+                    {
+                        if (item.StartsWith("package:"))
+                        {
+                            var query = '"' + $"{item.Replace("package:", "")}" + '"';
+                            queryStringBuilder.Append($" AND STRINGEQUALS(ARRAY_SLICE(r.Revisions, -1)[0].Files[0].PackageName, {query}, true)");
+                        }
+                        else if (item.StartsWith("author:"))
+                        {
+                            var query = '"' + $"{item.Replace("author:", "")}" + '"';
+                            queryStringBuilder.Append($" AND STRINGEQUALS(r.Author, {query}, true)");
+                        }
+                        else if (item.StartsWith("service:"))
+                        {
+                            var query = '"' + $"{item.Replace("service:", "")}" + '"';
+                            queryStringBuilder.Append($" AND STRINGEQUALS(r.ServiceName, {query}, true)");
+                        }
+                        else if (item.StartsWith("pr:"))
+                        {
+                            var query = '"' + $"{item.Replace("pr:", "")}" + '"';
+                            queryStringBuilder.Append($" AND ENDSWITH(ARRAY_SLICE(r.Revisions, -1)[0].Label, {query}, true)");
```

Comment:
This is okay for now. But relying on label to find PR based reviews is risky. Label filed format is something we might easily change in the future. And manual review can have any value as label.

Reply:
`testing code`